app/controllers/application_controller.rb

```diff
@@ -3,8 +3,17 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   include SessionsHelper
+	before_filter :current_user, :permission_check, :check_if_logged_on
 
-  before_filter :current_user, :permission_check, :check_if_logged_on
+	# addresses that can be accessed while not logged in, but should not be accessed once logged in
+	def outside_path?(param)
+		outside_addresses = [
+			"/login",
+			"/signup",
+			"/"
+		]
+		outside_addresses.any?{ |path| path[param] }
+	end
```

Comment:
The readability was what I was aiming to achieve, changing it back to the regular spacing indentation.